autokeras/hypermodel/resnet.py

```diff
@@ -0,0 +1,330 @@
+        name: string, stack label.
+
+    # Returns
+        Output tensor for the stacked blocks.
+    """
+    x = block3(x,
+               filters,
+               stride=stride1,
+               groups=groups,
+               name=name + '_block1')
+    for i in range(2, blocks + 1):
+        x = block3(x, filters, groups=groups, conv_shortcut=False,
+                   name=name + '_block' + str(i))
+    return x
+
+
+def resnet(stack_fn,
+           preact,
+           use_bias,
+           input_tensor=None,
+           pooling=None):
+    """Instantiates the ResNet, ResNetV2, and ResNeXt architecture.
+
+    # Arguments
+        stack_fn: a function that returns output tensor for the
+            stacked residual blocks.
+        preact: whether to use pre-activation or not
+            (True for ResNetV2, False for ResNet and ResNeXt).
+        use_bias: whether to use biases for convolutional layers or not
+            (True for ResNet and ResNetV2, False for ResNeXt).
+        input_tensor: optional Keras tensor
+            (i.e. output of `layers.Input()`)
+            to use as image input for the model.
+        pooling: optional pooling mode for feature extraction
+            when `include_top` is `False`.
+            - `None` means that the output of the model will be
+                the 4D tensor output of the
+                last convolutional layer.
+            - `avg` means that global average pooling
+                will be applied to the output of the
+                last convolutional layer, and thus
+                the output of the model will be a 2D tensor.
+            - `max` means that global max pooling will
+                be applied.
+
+    # Returns
+        The output node.
+    """
+    bn_axis = 3 if backend.image_data_format() == 'channels_last' else 1
+
+    x = layers.ZeroPadding2D(padding=((3, 3), (3, 3)),
+                             name='conv1_pad')(input_tensor)
+    x = layers.Conv2D(64,
+                      7,
+                      strides=2,
+                      use_bias=use_bias,
+                      name='conv1_conv')(x)
+
+    if preact is False:
+        x = layers.BatchNormalization(axis=bn_axis, epsilon=1.001e-5,
+                                      name='conv1_bn')(x)
+        x = layers.Activation('relu', name='conv1_relu')(x)
+
+    x = layers.ZeroPadding2D(padding=((1, 1), (1, 1)), name='pool1_pad')(x)
+    x = layers.MaxPooling2D(3, strides=2, name='pool1_pool')(x)
+
+    x = stack_fn(x)
+
+    if preact is True:
+        x = layers.BatchNormalization(axis=bn_axis, epsilon=1.001e-5,
+                                      name='post_bn')(x)
+        x = layers.Activation('relu', name='post_relu')(x)
+
+    if pooling == 'avg':
+        x = layers.GlobalAveragePooling2D(name='avg_pool')(x)
+    elif pooling == 'max':
+        x = layers.GlobalMaxPooling2D(name='max_pool')(x)
+
+    return x
+
+
+class ResNetBlock(hyper_block.HyperBlock):
```

Comment:
Could you simply use `kerastuner.applications.HyperResnet`?

Reply:
The kerastuner.applications.HyperResnet is with the Dense layers after the conv layers. So we have to change the HyperResnet in order to use it.